src/public/static/js/client/worker.ts

```diff
@@ -35,7 +36,30 @@ const run = async () => {
             return;
         }
 
-        return (await response.json()) as unknown;
+        const data = (await response.json()) as unknown;
+
+        if (
+            !(
+                typeof data == "object" &&
+                data != null &&
+                "doNotTerminate" in data &&
+                data.doNotTerminate
+            )
+        ) {
+            postMessage({
+                type: "terminateSettings",
+                ...setupData,
+                terminate: false
+            });
+        } else {
+            postMessage({
+                type: "terminateSettings",
+                ...setupData,
+                terminate: true
+            });
+        }
```

Comment:
This can be simplified.

```suggestion
        postMessage({
            type: "terminateSettings",
            ...setupData,
            terminate: !( data?.hasOwnProperty("doNotTerminate") && data.doNotTerminate )
        });
```